### PR TITLE
core: separate chunk path from PathProperties

### LINKED ISSUE #5242: core: kt-infra: stdcm: remove chunks from PathProperties
Chunks, beginOffset and EndOffset should not be accessed from PathProperties. Move them to PathPropertiesImpl, do not let them be accessed, and modify the rest of the code accordingly.

### PR DESCRIPTION
**Pull request by Erashin**
Fixes #5242.

### CODE REVIEW

**Review by axrolld — COMMENTED** [review]
Isn't there a problem in the commits of this PR ? I feel like i've already reviewed those changes in another PR 🤔

**Comment by eckter** [comment]
> Isn't there a problem in the commits of this PR ? I feel like i've already reviewed those changes in another PR 🤔

Only the last commit matters, the rest just means the branch needs to be rebased

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/PathPropertiesImpl.kt [hunk 1]
```diff
@@ -1,31 +1,29 @@
...
+import fr.sncf.osrd.utils.indexing.mutableDirStaticIdxArrayListOf
 import fr.sncf.osrd.utils.units.Distance
 import fr.sncf.osrd.utils.units.Offset
 import fr.sncf.osrd.utils.units.Speed
 import fr.sncf.osrd.utils.units.meters
 
-data class PathPropertiesImpl(
-    val infra: TrackProperties,
-    override val chunks: DirStaticIdxList<TrackChunk>,
+data class ChunkPath(
```
Comment (Khoyo) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/PathPropertiesImpl.kt:17]: Some doc would be nice

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/PathPropertiesImpl.kt [hunk 2]
```diff
@@ -255,3 +253,33 @@ fun getOffsetOfTrackLocationOnChunksOrThrow(
 ): Distance {
     return getOffsetOfTrackLocationOnChunks(infra, location, chunks) ?: throw RuntimeException()
 }
+
+@JvmName("buildChunkPath")
```
Comment (Khoyo) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/PathPropertiesImpl.kt:260]: Same, a comment about the fact the the returned path is only a subset of the given chunks would be nice

**Review by Khoyo — APPROVED** [review]
Otherwise LGTM

**Pull request merged**